### ISSUE TITLE
Redesign tags UX

### ISSUE THREAD

**Issue by majkinetor** [1Remote/1Remote#401]
Tags UX is needlessly large and non responsive: 

<details><summary>details</summary> 

![image](https://user-images.githubusercontent.com/85767/236676707-d7e0367a-7221-4624-8212-933296f6b463.png)

</details>

Tags UX should be completelly redesigned. 

## Redesign

1. [x] Remove # in the name, it just spams, tags have clear UI place on both lister and launcher
2. [x] Make list responsive (regression)
3. [x] Context option to close tag tab (right click on tab) 
4. [ ] Change tab order by drag and droping them (useful with #405 ALT + n)
5. [x] ~~Tag details with middle click~~
    ~~1. Tag description (to be shown in a tooltip)~~
    ~~2. Tag hotkey?~~
7. [x] Add pin tag as context menu item instead of using a checkbox which spams a lot and is used a little
8. [x] Tags should not be case sensitive
    - Currently tag lists shows both lowercase and uppercase tags with different counts
    - However, pinning them, you get the same view

**Comment by majkinetor** [comment]
BTW, this tag list not showing in columns is something that apperaed in 1.0.beta3, in 0.4 it was not like that.

**Comment by majkinetor** [comment]
Actually, I dont know what I was thinking when I sad that clicking on tag could be used as a checkbox, as clicking on tag is used to activate that tag. But nevertheless, this is simple to change, we could add "pin tag" in context menu which doesn't spoil the UX pinning is not that frequent operation.

Regarding number of columns, I think they should just be listed horizontally as much as it can be filled with current screen, there is no need to artificially introduce columns as you may have short and long tags. They could be sorted by name (like now) or frequency. This should be very easy to implement and maintain and it will be responsive relative to the screen size.

**Comment by majkinetor** [comment]
Mockup:

![image](https://user-images.githubusercontent.com/85767/236863223-f83575ec-7922-488d-920f-b1a7711b906e.png)

```
[ Tag details             x]

name:             tag1
description:      Foo bar baz

[Delete tag]      [Update tag]
```

**Comment by VShawn** [comment]
> Mockup:
> 
> ![image](https://user-images.githubusercontent.com/85767/236863223-f83575ec-7922-488d-920f-b1a7711b906e.png)
> 
> ```
> [ Tag details             x]
> 
> name:             tag1
> description:      Foo bar baz
> 
> [Delete tag]       [Pin tag] [Save]
> ```

new users may not know there's a right click menu for pin.

**Comment by majkinetor** [comment]
If that is your worry, we can add a message :)

![image](https://github.com/1Remote/1Remote/assets/85767/6a1ba824-2680-4e97-b415-68dc142d3b20)

**Comment by majkinetor** [comment]
@VShawn can you quickly fix regression in latest 2 versions? Tag list is unusable now.

**Comment by VShawn** [comment]
I won't working on this befor #409 is closed.

I working on #395, and then #409, and then this one :)

**Comment by majkinetor** [comment]
OK, that will take time then. Can you fix this quickly becuase I can't pin antyhing without too much stress any more :)

![image](https://github.com/1Remote/1Remote/assets/85767/5a91e1de-f3fb-4f7e-980d-f568498c56ae)

**Comment by VShawn** [comment]
The tag description seems not not to be very necessary, but the cost of it is high. 

For example, it requires designing a new tag table (the current tag only stores a string in connection json), and handling data synchronization of the tag table when add \ del \ rename tags, and resolving situations where there are tags with the same name but different descriptions in two databases.

Therefore, please reconsider this. If it must be done, please create a new ticket specifically for it.

And case insensitivity is rude. You should not decide for the user that they cannot use "ab" after using "AB". AFAIK green is color while Green is a name.

**Comment by majkinetor** [comment]
OK, tag table is natural thing to have, I didn't think you would just put tags in connections. Lets skip it.

> And case insensitivity is rude. 

Almost no tag system I know is case sensitive. There is nothing rude about it, its expected, **especially on Windows** where file names are not case sensitive. If that is so, then search has a bug - if I click tag name, i should see that exact tag name, not all the variants like now:

![image](https://github.com/1Remote/1Remote/assets/85767/2a7dbba0-8362-4c20-b4b4-a4d4ed877de7)

**Comment by VShawn** [comment]
1. Yes there is a problem within tag search because when I developing this feature, I wanted to enable fuzzy searching while also supporting case sensitivity.... no better solution was found.

2. and please consider another situation: there is a tag "abcd" in db1 and a tag "Abcd" in db2. Should we merge these two tags if the it is not case sensitive? And when displaying the tag, should it be displayed as "abcd" or "Abcd"?

I have considered case sensitivity when building the tag system from the beginning, but these 2 still bother me now. I have thought about convert all tags to lowercase, but as mentioned sometimes capitalization can change the meaning of a word.

**Comment by majkinetor** [comment]
> but as mentioned sometimes capitalization can change the meaning of a word.

Is this really relevant for something like tags?

> there is a tag "abcd" in db1 and a tag "Abcd" in db2. Should we merge these two tags if the it is not case sensitive?

What I do in my projects is lowercase them. Nobody ever complained. 

As an example from the wild, [Diigo](https://diigo.com) which has tags as basic mechanism (you can't really use it without it, so you hundreeds of them) lowercases them in tag list, and only keeps user input as it is:

- <details><summary>Search shows lower case variants always</summary>
    
    ![image](https://github.com/1Remote/1Remote/assets/85767/3de43d75-2f51-4b8a-82ed-983ced1658dd)</details>
- <details><summary>Its listed as it is input on the bookmark</summary>

    ![image](https://github.com/1Remote/1Remote/assets/85767/6881abec-73aa-4c8b-bbed-3a3c4b416b32)</details>
- <details><summary>All tags are listed lowercase in the tag list</summary>
    
    ![image](https://github.com/1Remote/1Remote/assets/85767/0748464d-aafb-429b-ae9c-3567609f1049)</details>

So you could go that way, its IMO good UX as tags are listed as input on most important place and the one that you see most often - the objects having them (connections in this case).

So to recap:

1. Keep user input as it is
2. Search case insensitive 
3. Show tag list lowercase

**Comment by VShawn** [comment]
> @VShawn, It would be good to consistently allow for context menu, not only in tags:
> 
> 
> 
> ![image](https://github.com/1Remote/1Remote/assets/85767/c2a7a367-7352-4aac-a8c5-60a07a6f32aa)
> 
> 
> 
> 

Yeap  I woring on this now, since the 'right click ed' hijacked the mouse right click event, i need to refact this part to make context menu work

**Comment by majkinetor** [comment]
In latest nightly some tags do not work.
